contracts/ClaimsManager.sol

```diff
@@ -242,35 +239,65 @@ contract ClaimsManager is
         );
     }
 
-    function createClaim(
+    function upgradePolicy(
+        address claimant,
         address beneficiary,
         uint256 coverageAmountInUsd,
         uint256 claimsAllowedFrom,
         uint256 claimsAllowedUntil,
         string calldata policy,
+        string calldata metadata
+    ) external onlyManagerOrPolicyCreator returns (bytes32 policyHash) {
+        policyHash = keccak256(
+            abi.encodePacked(claimant, beneficiary, claimsAllowedFrom, policy)
```

Comment:
`policy` is the ipfs url, right?

Reply:
wouldn't it be more convenient (cheaper) to call this function with the `policyHash` as an arg?

Reply:
> policy is the ipfs url, right?

Yes, something like `/ipfs/Q...`

> wouldn't it be more convenient (cheaper) to call this function with the policyHash as an arg?

I did this for the `indexed claimant` in `UpgradedPolicy`. Otherwise @mcoetzee will have to fetch logs for each policy to check if it has been upgraded. Policy upgrades are rare events so it's not super important imo (though this many arguments caused a stack size error in https://github.com/api3dao/claims-manager/pull/45 but I think I'll be able to work around that).

Reply:
Ah yes this `event UpgradedPolicy` should be useful for the frontend. I guess in the case where the policy gets upgraded multiple times, the `UpgradedPolicy` event with the largest coverage would be the most recent one?

Reply:
The most recent policy is the currently valid one, not necessarily the largest one. It could go like
- Upgraded to $1000 coverage
- Received a $1000 payout
- Upgraded to $500 coverage

Reply:
Hmm I'm not sure if I understand which would be the currently valid one, as it seems `claimsAllowedUntil` doesn't need to be extended (it looks like we accept a value that is equal to `policyState.claimsAllowedUntil`)?
```solidity
        require(
            policyState.claimsAllowedUntil <= claimsAllowedUntil,
            "Policy allows claims for longer"
        );
```

Reply:
If a policy has an `UpgradedPolicy` event associated with it, you should use the `claimsAllowedUntil` of the latest `UpgradedPolicy` event. You probably don't want to use the `coverageAmountInUsd` field of the event to learn the coverage because there may have been payouts since then, so you would call `policyHashToState()` instead. That also returns you `claimsAllowedUntil` of the policy, so at that point you don't need `UpgradedPolicy` and maybe this should be called with `policyHash` as @acenolaza suggested above.

Reply:
Yeah the easy option would be to make a single multicall to get this dynamic data for all the user's policies:
```ts
  const calls = createdEvents.map((ev) => {
    return contract.interface.encodeFunctionData('policyHashToState', [ev.args.policyHash]);
  });
  const encodedResults = await contract.callStatic.multicall(calls);
  const decodedResults = encodedResults.map((res) => {
    return contract.interface.decodeFunctionResult('policyHashToState', res);
  });
```
But I am slightly concerned about users with a large number of policies, whether it would take long to load or not.

Reply:
I suppose getting `claimsAllowedUntil` from `UpgradedPolicy` is easy enough and you can build the current `coverageAmountInUsd` from the latest `UpgradedPolicy` event and the following payout events, then you wouldn't need to do any static calls. I'll leave this as is for `claimant` to be indexed for this to be possible.